source/adios2/core/IO.cpp

```diff
@@ -765,7 +763,7 @@ Engine &IO::Open(const std::string &name, const Mode mode, MPI_Comm mpiComm)
 
 Engine &IO::Open(const std::string &name, const Mode mode)
 {
-    return Open(name, mode, m_ADIOS.GetComm().AsMPI());
+    return Open(name, mode, m_ADIOS.GetComm().Duplicate());
```

Comment:
Rather than an explicit `Duplicate`, would it instead make sense to pass the comm object by value and have `Duplicate` be implemented in the copy constructor?

Reply:
`helper::Comm` is explicitly move-only and has no copy constructor.  This avoids hiding non-trivial operations like `MPI_Comm_dup` behind simple-looking syntax like a (possibly unintentional) copy.